File: Exercise_1.py
# Time Complexity: 
#   push: O(1)
#   pop: O(1)
#   peek: O(1)
# Space Complexity: O(n)
# Did this code successfully run on Leetcode :No
# Any problem you faced while coding this : No


# Your code here along with comments explaining your approach
# We implement a stack using a fixed-size array of MAX size.
# 'top' keeps track of the index of the last inserted element.
# When the stack is empty, top = -1.
class myStack:
  #Please read sample.java file before starting.
  #Kindly include Time and Space complexity at top of each file
     def __init__(self):
       self.stack = []
       self.stackSize = 0
        
     def isEmpty(self):
       return self.stackSize==0
         
     def push(self, item):
       self.stack.append(item)
       self.stackSize += 1
         
     def pop(self):
        if self.isEmpty():
          return None
        self.stackSize -= 1
        return self.stack.pop()
        
        
     def peek(self):
       if self.isEmpty():
         return None
       return self.stack[-1]
        
     def size(self):
       return self.stackSize
         
     def show(self):
       return self.stack
         

s = myStack()
s.push('1')
s.push('2')
print(s.pop())
print(s.show())

File: Exercise_1.java
// Time Complexity: 
//   push: O(1)
//   pop: O(1)
//   peek: O(1)
// Space Complexity: O(n)
// Did this code successfully run on Leetcode :No
// Any problem you faced while coding this : No


// Your code here along with comments explaining your approach
// We implement a stack using a fixed-size array of MAX size.
// 'top' keeps track of the index of the last inserted element.
// When the stack is empty, top = -1.

class Stack { 
    static final int MAX = 1000; 
    int top; 
    int a[] = new int[MAX]; // stack array

    // Constructor
    Stack() { 
        top = -1;  // stack is empty
    } 
  
    // Check if stack is empty
    boolean isEmpty() { 
        return (top < 0);
    } 

    // Push element onto stack
    boolean push(int x) { 
        if (top >= MAX - 1) {   // stack overflow check
            System.out.println("Stack Overflow");
            return false;
        }
        a[++top] = x;  // increment top and add element
        return true;
    } 
  
    // Pop element from stack
    int pop() { 
        if (top < 0) {  // underflow check
            System.out.println("Stack Underflow");
            return 0;
        }
        return a[top--]; // return top element and decrease top
    } 
  
    // Return top element without removing
    int peek() { 
        if (top < 0) {
            System.out.println("Stack Underflow");
            return 0;
        }
        return a[top];
    } 
} 
  
// Driver code 
class Main { 
    public static void main(String args[]) { 
        Stack s = new Stack(); 
        s.push(10); 
        s.push(20); 
        s.push(30); 
        System.out.println(s.pop() + " Popped from stack"); 
    } 
}
